Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -2410,14 +2410,7 @@ checkSetupVersions pkg =
     ]
   where
     criticalPkgs = ["Cabal", "base"]
-    deps = case typicalPkg pkg of
-      Right (pkgs', _) ->
-        Map.fromListWith intersectVersionRanges
-          [ (pname, vr)
-          | sbi <- maybeToList $ setupBuildInfo pkgs'
-          , Dependency pname vr _ <- setupDepends sbi
-          ]
-      _ -> Map.empty
+    deps = toDependencyVersionsMap ((=<<) setupDepends . maybeToList .  setupBuildInfo) pkg
```

Comment:
Section preferred: `((setupDepends =<<) . maybeToList . setupBuildInfo)`
Or, even better: `maybe [] setupDepends . setupBuildInfo`
Yeah, why turn `Maybe` into a 0-1 list and then use `concatMap`?

(Thought about: `=<<`:
That said, I am not a great fan of using `=<<` for lists...
The original list comprehension looked much more readable than the reformulation with `=<<` and `do`.
(But I acknowledge you wanted to factor out common code.))

Reply:
I used `foldMap setupDepends . setupBuildInfo`

I only touched this originally because I was looking at the wrong function somhow, and then I realized these were incredibly similar so I factored it out :sweat_smile: 
I don't particularly favor bind or list comprehensions in any case,
I just hadn't see list comprehensions used that much before in haskell code, so part of my understanding process was changing it into it's bind based representation.
>>= on lists can go really unexpected because it does cartesian products by default instead of pointwise products like ziplist does

Reply:
Thanks for fixing!